Bottom: bb373dec03a23765c5924386b8b621fc2f36679e
Top:    1d64c0ad0821d2fbf215518388675ce707d2e01a
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 15:09:01 +0800

autofs-5.0.7 - fix possible use after free in lookup_dir.c:lookup_init()

Add a missing error return in lookup_dir.c:lookup_init().


---

diff --git a/modules/lookup_dir.c b/modules/lookup_dir.c
index 07471b7..cbeda1f 100644
--- a/modules/lookup_dir.c
+++ b/modules/lookup_dir.c
@@ -98,6 +98,7 @@ int lookup_init(const char *mapfmt, int argc, const char *const *argv, void **co
 		free(ctxt);
 		warn(LOGOPT_NONE, MODPREFIX
 		     "dir map %s, is not a directory", argv[0]);
+		return 1;
 	}
 
 	*context = ctxt;
